### PR TITLE
Add Python 3.6 runtime test to Travis CI

### PR DESCRIPTION
**Pull request by matthewfeickert**
Add Python 3.6 runtime to the Travis CI testing suite.

### CODE REVIEW

**Comment by cdeil** [comment]
@matthewfeickert - Thanks!

👍 to add this build. See discussion in #332.

@HDembinski - Assigning to you to decide.

**Comment by matthewfeickert** [comment]
@cdeil @HDembinski Ah, I see that I may have misunderstood something about your project. You appear to already be testing Python 3.6 on AppVeyor. Also the CI only ran on AppVeyor, and not Travis.

As you don't have a `CONTRIBUTING.md`, can you explain to me how your testing scheme works? That is, why you test different versions on different CI systems? I just want to understand if this PR is useless or not for the project.

**Comment by cdeil** [comment]
@HDembinski - Looks like the travis-ci setup for iminuit is misconfigured.

This PR didn't trigger builds on travis-ci!?

And at  https://github.com/scikit-hep/iminuit#iminuit there's a link to your personal https://travis-ci.com/HDembinski/iminuit

@matthewfeickert - I haven't worked on iminuit for a while, and actually I'm unsubscribing again now. @HDembinski will sort out this CI config issue and reply to your issue #332 .

**Comment by matthewfeickert** [comment]
Fixed it. [Turns out that](https://docs.travis-ci.com/user/pull-requests/)

> Please note, Draft Pull Requests are not currently supported on Travis CI.

I renamed it as a WIP PR and then Travis started up. :+1:

**Comment by matthewfeickert** [comment]
Hi @HDembinski. It seems that the [`iminuit` pypy tests are broken on Travis CI](https://travis-ci.org/scikit-hep/iminuit/jobs/528543543a) (unrelated to the point of this PR). Is this already known to you? If not, I'll open an Issue (I just didn't want to spam you if you're already aware and working on it).

**Comment by HDembinski** [comment]
@matthewfeickert Sorry for the late reply, I am busy with other projects.

I see that the pypy test is broken. I am not sure what going on, the last build of the develop branch was fine.
https://travis-ci.com/HDembinski/iminuit/builds/104927830

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,5 @@
 language: python
 
-sudo: false
```
Comment (HDembinski) [inline .travis.yml]: Why did you remove sudo: false? Please revert.

**Comment by HDembinski** [comment]
@cdeil 

> @HDembinski - Looks like the travis-ci setup for iminuit is misconfigured.
> 
> This PR didn't trigger builds on travis-ci!?

Nothing is misconfigured, see what @matthewfeickert wrote later.

> And at https://github.com/scikit-hep/iminuit#iminuit there's a link to your personal https://travis-ci.com/HDembinski/iminuit

Right, that should be updated.

File: .travis.yml (hunk 1 above)
Comment (matthewfeickert) [reply .travis.yml]: > Why did you remove sudo: false? Please revert.

In [November 2018 Traivs announced they would be retiring this API](https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration). In the post they advise that

> If you currently specify `sudo: false` in your `.travis.yml`, we recommend removing that configuration soon.

**Comment by matthewfeickert** [comment]
@HDembinski I've rebased this to pickup your more recent changes and while the pypy build still fails the CI passes now that you've added https://github.com/scikit-hep/iminuit/commit/d8c8ddbe8fb731473d31a85ddc4c5f61d96d36a6. So this is ready for review now.

File: .travis.yml (hunk 1 above)
Comment (HDembinski) [reply .travis.yml]: Thank you for noting this, but could you still please revert this, as it is not related to change you want to make? There should be only one feature change per PR, and these are two unrelated feature changes.

Comment (HDembinski) [reply .travis.yml]: I will remove the `sudo: false` later.

Comment (matthewfeickert) [reply .travis.yml]: @HDembinski Sure. I'll do so and rebase this. I totally appreciate the idea of wanting PRs to be as atomic as possible &mdash; the only reason I didn't do that here is it felt like an incredibly minor change given that the API is being dropped and not modified and so will have downstream effects.

**Comment by matthewfeickert** [comment]
@HDembinski I've rebased to drop the last commit which addresses your request RE: the dropped `sudo: false` API. Let me know if there is anything further you'd like to see here. :+1:

**Comment by HDembinski** [comment]
Thank you very much for this patch!
